ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala

```diff
@@ -402,9 +402,9 @@ private[ember] object H2Client {
       RequestKey(uri.scheme.getOrElse(Scheme.http), authOrAddr)
     }
 
-    def getAddress[F[_]: Sync](
+    def getAddress[F[_]](
         requestKey: RequestKey
-    ): F[Either[UnixSocketAddress, SocketAddress[Host]]] =
+    )(implicit F: MonadThrow[F]): F[Either[UnixSocketAddress, SocketAddress[Host]]] =
```

Comment:
Since this is internal code, can we make the change in series/0.23 and merge up? Thanks.

Reply:
> Since this is internal code, can we make the change in series/0.23 and merge up?

We would have to add MiMa filters in that case, I guess.

Reply:
From my point of view, it's just a non-user-facing clean-up... But alright, I do backport if this comment will get a few likes 😄

Reply:
@danicheg exactly. IMO whenever we deliberately make merge conflicts (as this PR is doing) we have to evaluate whether (1) it's fixing bugs or (2) its fixing public APIs.

If a PR is not doing either of those things, we should have a good reason as to why we are going to create merge conflicts for every future merge that has touched this code, considering the fact that merge conflicts are often resolved by someone who is not an expert in the conflicting code and that even for someone who does know what they are doing it is very easy to make mistakes and lose changes.

Reply:
Merge conflicts are a fair point. Disregarding I think merge conflicts for this particular case should be crystal clear for anyone at least a little experienced in `cats`, I will do a follow-up for 0.23-series.

Reply:
> Disregarding I think merge conflicts for this particular case should be crystal clear for anyone at least a little experienced in `cats`

Sure, but that is only one half of the merge conflict :) merge conflicts happen when two independent changes are made to the same code. Even though this first change may be simple, the second change the causes the conflict may be more complicated. Whoever resolves the conflict must navigate both of those changes.

> I will do a follow-up for 0.23-series.

Thank you.